src/api/app/controllers/webui/patchinfo_controller.rb

```diff
@@ -312,13 +312,14 @@ def get_binaries
 
   def require_exists
     unless params[:package].blank?
-      @package = Package.get_by_project_and_name(@project.to_param, params[:package], use_source: false)
+      @package = Package.get_by_project_and_name(params[:project], params[:package], use_source: false)
     end
-    @patchinfo = @file = @package.patchinfo
 
-    unless @file
+    unless @package && @package.patchinfo
+      # FIXME: should work for remote packages
       flash[:error] = "Patchinfo not found for #{params[:project]}"
       redirect_to controller: 'package', action: 'show', project: @project, package: @package and return
     end
+    @patchinfo = @file = @package.patchinfo
```

Comment:
it is still used in the views (at least test suite breaks when trying to remove it, not looked deeper)